File: content/code-examples/tested/java/driver-sync/timeseries/Limitations.snippet.agg-pipeline-for-distinct.java
collection.createIndex(Indexes.compoundIndex(
        Indexes.ascending("meta.project"),
        Indexes.ascending("meta.type")
));

var pipeline = Arrays.asList(
        Aggregates.match(Filters.eq("meta.project", 10)),
        Aggregates.group("$meta.type")
);
List<Document> result = collection.aggregate(pipeline).into(new ArrayList<>());

File: content/code-examples/tested/javascript/driver/time-series/limitations.snippet.agg-pipeline-for-distinct.js
await collection.createIndex({ 'meta.project': 1, 'meta.type': 1 });

const pipeline = [
  { $match: { 'meta.project': 10 } },
  { $group: { _id: '$meta.type' } },
];

const result = await collection.aggregate(pipeline).toArray();
